### ISSUE TITLE
save_model does not retain scheduler function

### ISSUE THREAD

**Issue by tmontana** [dreamquark-ai/tabnet#211]
**Describe the bug**

model parameters change after save and re-load
**What is the current behavior?**
after training one_clf:

print(one_clf)

TabNetClassifier(n_d=16, n_a=16, n_steps=5, gamma=1.25, cat_idxs=[], cat_dims=[], cat_emb_dim=1, n_independent=2, n_shared=2, epsilon=1e-15, momentum=0.3, lambda_sparse=0.0001, seed=42, clip_value=2.0, verbose=1, optimizer_fn=<class 'torch.optim.adam.Adam'>, optimizer_params={'lr': 0.02}, scheduler_fn=<class 'torch.optim.lr_scheduler.StepLR'>, scheduler_params={'gamma': 0.95, 'step_size': 10}, mask_type='entmax', input_dim=228, output_dim=5, device_name='cuda')

saved_name=one_clf.save_model('1_test_model')
clf_2.load_model(saved_name)

print(clf_2)

TabNetClassifier(n_d=16, n_a=16, n_steps=5, gamma=1.25, cat_idxs=[], cat_dims=[], cat_emb_dim=1, n_independent=2, n_shared=2, epsilon=1e-15, momentum=0.3, lambda_sparse=0.0001, seed=42, clip_value=2.0, verbose=1, optimizer_fn=<class 'torch.optim.adam.Adam'>, optimizer_params={'lr': 0.02}, scheduler_fn=None, scheduler_params={'gamma': 0.95, 'step_size': 10}, mask_type='entmax', input_dim=228, output_dim=5, device_name='cuda')

scheduler_fn is now None

**If the current behavior is a bug, please provide the steps to reproduce.**

see above

**Expected behavior**

The models should be exactly the same before and after saving and loading

**Screenshots**

**Other relevant information:**
poetry version:  
python version: 
Operating System: 
Additional tools:

**Additional context**

version 2.0.1

**Comment by Optimox** [comment]
Well we are not saving the scheduler because saving the model is mostly used in order to make predictions after training your model, I don't see why you would need to keep the scheduler.

If you want to retrain you can just define a new scheduling scheme. Why can't you just define a new one if you need one?

**Comment by tmontana** [comment]
No worries - I'm used to pickling models so they retain 100% of the same info as during training and wanted to make sure this was expected behavior.  closing now.  Thanks
